Mathlib/CategoryTheory/Quotient.lean

```diff
@@ -67,6 +67,30 @@ theorem comp_right {a b c : C} (g : b ⟶ c) :
     ∀ (f₁ f₂ : a ⟶ b) (_ : CompClosure r f₁ f₂), CompClosure r (f₁ ≫ g) (f₂ ≫ g)
   | _, _, ⟨x, m₁, m₂, y, h⟩ => by simpa using CompClosure.intro x m₁ m₂ (y ≫ g) h
 
+theorem CompClosure.congruence :
```

Comment:
It would be nice to prove this after `functor r` is defined, because it seems this could then be deduced from the fact that `Relation.EqvGen (@CompClosure C _ r X Y)` is, by the basic properties of quotients, the equivalence relation on `X ⟶ Y` such that `f` and `g` are related iff `(functor r).map f = (functor r).map g`. In general, we could also introduce `Functor.homRel` which sends a functor `F` from `C` to `D` to the relation on morphisms on `C` defined by `F.map f = F.map G`, and then we can prove `Congruence F.homRel` (all the fields can be obtainted mostly `by aesop`). Then, up to a `rw`, this result would be the particular case `Congruence (functor r).homRel`.

Reply:
@joelriou I believe I've now implemented all of your suggestions. See

+ Functor.homRel
+ Functor.homRelCongruence
+ functorHomRel_eq_compClosureEqvGen (better name?)
+ compClosure.congruence (I thought that perhaps this should have been lowercase, but maybe that is a mistake)